design/anoncreds.md

```diff
@@ -0,0 +1,507 @@
+    * Issuer signs (by his Cred Def's public key) the `issuanceTime` and sends it to the pool.
+    * Each node verifies that `issuanceTime` is not less that the current one, and signs the result with BLS key.
+    * Each node then sends the signed result to the Issuer (no need to write anything to the ledger).
+    * The issuer prepares a BLS multi-signature (making sure that there is a consensus)
+    and adds the BLS-signed proof of the `issuanceTime` to the credential.
+    * The verifier will then use the proof to make sure that the `issuanceTime` is really the correct one.
+* The `issuanceTime` needs to be verified in each proof.
+    * The Verifier should use Predicates (instead of disclosing) for the value of `issuanceTime`
+    to avoid correlation. 
+    * It's possible also to disclose `issuanceTime`, but we don't force it.
+    * If it's not disclosed and not verified as a Predicate, then there is a chance the the proof verification will fail because 
+of key rotations, since the latest keys will be used.
+
+
+
+### SCHEMA
+
+#### SCHEMA txn
+```
+{
+    "data": {
+        "uuid":"GEzcdDLhCpGCYRHW82kjHd",
+        "attrNames": ["undergrad","last_name","first_name","birth_date","postgrad","expiry_date"],
+        "name":"Degree",
+        "version":"1.0",
+    },
+    
+    "reqMetadata": {
+        "issuerDid":"L5AD5g65TDQr1PPHHRoiGf",
+        .....
+    },
+    
+....
+}
+```
+`uuid` is Schema's UUID. It's different from `issuerDid`.
+
+#### Restrictions
+
+* Existing Schema (identified by the Schema `uuid`) can be modified/changed/evolved.
+* Only the `issuerDid` who created the Schema can modify it (that is we need to keep the ownership).
+* It's not possible to create multiple entities with the same `uuid` (so, it's unique within the ledger). 
+
+#### State
+
+We need to have two records for Schema in State Trie in order to have
+1. Simple referencing of Schemas in the protocol (by Schema DID)
+1. Requirements 8
+
+Record 1:
+* key: `schemaIssuerDid | SchemaMarker | schemaName | schemaVersion | schemaUUID` 
+* value: aggregated txn data
+
+Record 2:
+* key: `schemaUUID`
+* value: Record 1 key
+
+
+#### GET_SCHEMA
+```
+{
+    'data': {
+        'uuid': 'GEzcdDLhCpGCYRHW82kjHd',
+    },
+...
+}
+```
+1. Lookup State Trie to get `schemaIssuerDid | SchemaMarker |schemaName | schemaVersion | schemaUUID` by `uuid`
+using Record2. 
+1. Lookup State Trie to get data by the key found above (Record1).
+
+So, we will have 2 lookups for each request. 
+
+### CRED_DEF
+
+The Definition of credentials for the given Schema by the given Issuer.
+It contains public keys (primary and revocation),
+signature type and reference to the Schema. 
+
+#### CRED_DEF txn
+```
+{
+    "data": {
+        "uuid":"TYzcdDLhCpGCYRHW82kjHd",
+        "publicKeys": {},
+        "schemaRef":"GEzcdDLhCpGCYRHW82kjHd",
+        "signatureType":"CL",
+        "oldKeyTrustTime": {                    (optional)
+            "from": "10", 
```

Comment:
What are `10` and `30`?

Reply:
timestamps :) maybe not the good values, but just for simplicity.